packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -0,0 +1,50 @@
+import * as cdk from '@aws-cdk/core';
+import { TagUpdateConstraintOptions } from '../constraints';
+import { IPortfolio } from '../portfolio';
+import { IProduct } from '../product';
+import { CfnPortfolioProductAssociation, CfnResourceUpdateConstraint } from '../servicecatalog.generated';
+import { hashValues } from './util';
+import { InputValidator } from './validation';
+
+export class AssociationManager {
+  public static associateProductWithPortfolio(
+    portfolio: IPortfolio, product: IProduct,
+  ): { associationKey: string, cfnPortfolioProductAssociation: CfnPortfolioProductAssociation } {
+    const associationKey = hashValues(portfolio.node.addr, product.node.addr, portfolio.stack.node.addr);
```

Comment:
I don't get why are you using `portfolio.stack` here, and not `product.stack`? You're adding the `CfnPortfolioProductAssociation` under the `portfolio` object in this method. So `portfolio.stack` doesn't change anything here (it doesn't add any variance to the uniqueness of the ID), because it will always be the same.

Reply:
I think I understand what you are saying.  In that case, by adding the l1 underneath the portfolio for all calls, and using the product stack name, we will have unique IDS for valid associations even if multiple stacks.  I think before we had used the portfolio stack which is why I added it here.